code/game/machinery/constructable_frame.dm

```diff
@@ -281,41 +288,90 @@
 					replacer.play_rped_sound()
 				return
 
-			if(isitem(P) && get_req_components_amt())
-				for(var/I in req_components)
-					if(istype(P, I) && (req_components[I] > 0))
-						if(isstack(P))
-							var/obj/item/stack/S = P
-							var/used_amt = min(round(S.get_amount()), req_components[I])
-
-							if(used_amt && S.use(used_amt))
-								var/obj/item/stack/NS = locate(S.merge_type) in components
-
-								if(!NS)
-									NS = new S.merge_type(src, used_amt)
-									components += NS
-								else
-									NS.add(used_amt)
-
-								req_components[I] -= used_amt
-								to_chat(user, span_notice("You add [P] to [src]."))
-							return
-						if(!user.transferItemToLoc(P, src))
-							break
+			for(var/stock_part_base in req_components)
+				var/stock_part_path
+
+				if (ispath(stock_part_base, /obj/item))
+					stock_part_path = stock_part_base
+				else if (ispath(stock_part_base, /datum/stock_part))
+					var/datum/stock_part/stock_part_datum_type = stock_part_base
+					stock_part_path = initial(stock_part_datum_type.physical_object_type)
+				else
+					stack_trace("Bad stock part in req_components: [stock_part_base]")
+					continue
+
+				if (req_components[stock_part_path] == 0)
+					continue
+
+				if (!istype(P, stock_part_path))
+					continue
+
+				if(isstack(P))
+					var/obj/item/stack/S = P
```

Comment:
single letter var ☹️

Reply:
Not mine!

Reply:
Nerd!
![image](https://user-images.githubusercontent.com/3625094/205473486-c94d671e-25db-4268-8abd-e064321d9f8e.png)

Reply:
When you're already touching the diff, you should also change the variable name. You'll notice I did the same.

When you're just adding code, changing all the variable names makes for a harder to read diff, and it's okay to let it aside unless a maintainer requires it.

Reply:
Be the change you want to see!

Reply:
The change I want to see is PR diffs that are easy to read, this proc is huge :)